### PR TITLE
Docs: clarify failure situations in Azure

### PR DESCRIPTION
**Pull request by rstacruz**
It seems the docs failed to mentioned that you will need an SSH keypair.

Also, I added some clarifications on what to do in case provisioning fails.

### CODE REVIEW

**Comment by rstacruz** [comment]
It also seems like `publicDomainName` is required.. might want to document that.

**Comment by monaka** [comment]
@rimusz I also got deployment failures caused by publicDomainName. It must be required.

**Comment by monaka** [comment]
I made the pull request #4856 to fix ``publicDomainName`` issue.

File: docs/installing_deis/azure.rst [hunk 1]
```diff
@@ -70,16 +74,14 @@ Generate a new discovery URL for the deployment so the hosts can find each other
...
+cluster. Any of the values in ``parameters.json`` are defaults and can be customized if desired.
 
-.. code-block:: console
+- For ``sshKeyData``, use the contents of ``~/.ssh/deis.pub``, which should only be a single line.
 
-    $ base64 azure-user-data
+- For ``customData``, you'll need to supply the base64-encoded version of ``azure-user-data``. This
+  can be generated using the command ``base64 azure-user-data``.
 
-Paste the result into ``parameters.json``. Any of the values in ``parameters.json``
```
Comment (bacongobbler) [inline docs/installing_deis/azure.rst]: is this no longer relevant?

**Pull request closed without merging**